soc: st: common: Rename STM32_PWR_WKUP_PIN_SRC_x

jurenat opened the pull request:
This renames the STM32_PWR_WKUP_PIN_SRC_x symbols to better match their meaning. It also adds a new symbol (STM32_PWR_WKUP_LINE_NOT_MUXED) for SoCs without wake-up mux support.

jurenat commented on include/zephyr/dt-bindings/power/stm32_pwr.h:
```diff
@@ -17,9 +17,10 @@
+#define STM32_PWR_WKUP_LINE_0         0
+#define STM32_PWR_WKUP_LINE_1         1
+#define STM32_PWR_WKUP_LINE_2         (1 << 2)
+#define STM32_PWR_WKUP_LINE_NOT_MUXED STM32_PWR_WKUP_LINE_0
```
It looked strange to me, but I did not check with RM. Fixed.

mathieuchopstm replied:
These values don't match anything in RM, they are for usage by the driver only - see: https://github.com/zephyrproject-rtos/zephyr/blob/8f2f97ce320ae184e3b2f572436a3758580205fe/soc/st/stm32/common/stm32_wkup_pins.c#L226-L237

mathieuchopstm commented on include/zephyr/dt-bindings/power/stm32_pwr.h:
```diff
@@ -17,9 +19,11 @@
+/* Use this value on series where wake-up line source is fixed/not configurable */
+#define STM32_PWR_WKUP_LINE_NOT_MUXED STM32_PWR_WKUP_LINE_0
+#define STM32_PWR_WKUP_LINE_0         BIT(0)
+#define STM32_PWR_WKUP_LINE_1         BIT(0)
```
```suggestion
#define STM32_PWR_WKUP_LINE_1         BIT(1)
```

mathieuchopstm reviewed (commented):
Thanks for thinking about C0 :slightly_smiling_face:

Could you also update the comment at the top of the file?
https://github.com/zephyrproject-rtos/zephyr/blob/b53937239f18f708ab20dd927458d80a8b4ce39a/dts/arm/st/c0/stm32c0.dtsi#L12-L22

jurenat commented:
> Thanks for thinking about C0 🙂
> 
> Could you also update the comment at the top of the file?
> 
> https://github.com/zephyrproject-rtos/zephyr/blob/b53937239f18f708ab20dd927458d80a8b4ce39a/dts/arm/st/c0/stm32c0.dtsi#L12-L22

Did I miss something? The comment at the beginning of the stm32c0.dtsi file is updated in [7e1e7ec](https://github.com/zephyrproject-rtos/zephyr/pull/88242/commits/7e1e7ec7729cd37cbfceb8116d56c6aebff5a80b)

JarmouniA reviewed (commented):
I see a benefit in having a separate flag for non-muxed wake-up pins.
About the renaming, I don't see why it's necessary or how the new name is clearer. BTW in the ref man they are called Wake-up Sources.
![IMG](https://github.com/user-attachments/assets/c38159b4-341b-41f3-a608-c19a79050989)

mathieuchopstm commented:
> > Thanks for thinking about C0 🙂
> > Could you also update the comment at the top of the file?
>
> Did I miss something? The comment at the beginning of the stm32c0.dtsi file is updated in [7e1e7ec](https://github.com/zephyrproject-rtos/zephyr/pull/88242/commits/7e1e7ec7729cd37cbfceb8116d56c6aebff5a80b)

Indeed :facepalm: 

Sorry for the unnecessary comment, I didn't see it.

mathieuchopstm commented:
> I see a benefit in having a separate flag for non-muxed wake-up pins. About the renaming, I don't see why it's necessary or how the new name is clearer. BTW in the ref man they are called Wake-up Sources.

Indeed; I had originally suggested `STM32_PWR_WKUP_LINE_SRC_x` but it flew over my mind when reviewing. We definitely want the `SRC` part, otherwise it is very confusing.

We could also name the defines `WKUP_EVENT_SRC_x`/`WKUP_EVT_SRC_x` instead of `WKUP_LINE` to stick to U5 RefMan more closely (IMO, all three options are fine).

What do you think @JarmouniA ?

jurenat commented:
> > I see a benefit in having a separate flag for non-muxed wake-up pins. About the renaming, I don't see why it's necessary or how the new name is clearer. BTW in the ref man they are called Wake-up Sources.
> 
> Indeed; I had originally suggested `STM32_PWR_WKUP_LINE_SRC_x` but it flew over my mind when reviewing. We definitely want the `SRC` part, otherwise it is very confusing.
> 
> We could also name the defines `WKUP_EVENT_SRC_x`/`WKUP_EVT_SRC_x` instead of `WKUP_LINE` to stick to U5 RefMan more closely (IMO, all three options are fine).
> 
> What do you think @JarmouniA ?

Oh, this was my mistake. I misread the name in the PR you suggested this.

If you both agree, I will change the names to `WKUP_EVT_SRC_x`.

mathieuchopstm commented:
> If you both agree, I will change the names to `WKUP_EVT_SRC_x`.

@JarmouniA said OK and I agree as well, so you can go for it :slightly_smiling_face:

jurenat commented on include/zephyr/dt-bindings/power/stm32_pwr.h:
```diff
@@ -17,9 +19,11 @@
-#define STM32_PWR_WKUP_PIN_SRC_1	1
-#define STM32_PWR_WKUP_PIN_SRC_2	(1 << 2)
+/* Use this value on series where wake-up line source is fixed/not configurable */
+#define STM32_PWR_WKUP_LINE_NOT_MUXED STM32_PWR_WKUP_LINE_0
```
So there should be two types of symbols? Isn't it confusing?

```
#define STM32_PWR_WKUP_PIN_NOT_MUXED WKUP_EVT_SRC_0
#define WKUP_EVT_SRC_0         0
#define WKUP_EVT_SRC_1         1
#define WKUP_EVT_SRC_2         (1 << 2)
```

mathieuchopstm replied:
`Wake-up event` is the name used by RM0456 though
![Screenshot from 2025-04-09 14-48-29](https://github.com/user-attachments/assets/e645158b-e942-4efc-912d-c6d7aed7a34c)
(which doesn't mean other things can't be wake-up events)

mathieuchopstm commented on include/zephyr/dt-bindings/power/stm32_pwr.h:
```diff
@@ -17,9 +19,11 @@
-#define STM32_PWR_WKUP_PIN_SRC_0	0
-#define STM32_PWR_WKUP_PIN_SRC_1	1
-#define STM32_PWR_WKUP_PIN_SRC_2	(1 << 2)
+/* Use this value on series where wake-up line source is fixed/not configurable */
```
IMO `value` is fine as only one of these defines should be used at a time (i.e., `(EVT_SRC_0 | EVT_SRC_1)` is an invalid usage).

(Using bitwise logic isn't really necessary for this reason, but w/e...)

```suggestion
/* Use this value on series where wake-up event source is fixed/not configurable */
```

jurenat committed:
soc: st: common: Rename STM32_PWR_WKUP_PIN_SRC_x

This renames the STM32_PWR_WKUP_PIN_SRC_x symbols to better match
their meaning. It also adds a new symbol (STM32_PWR_WKUP_PIN_NOT_MUXED)
for SoCs without wake-up mux support.

Signed-off-by: Tomáš Juřena <jurenatomas@gmail.com>

jurenat committed:
include: dt-bindings: power: stm32_pwr: Use BIT macro

Use BIT to create valid bitfields for wakeup sources.

Signed-off-by: Tomáš Juřena <jurenatomas@gmail.com>

martinjaeger reviewed (approved):
Looks good and thanks for fixing this.
